projects/angular/src/forms/datepicker/date-container.ts

```diff
@@ -175,6 +178,36 @@ export class ClrDateContainer extends ClrAbstractContainer implements AfterViewI
     );
   }
 
+  /**
+   * Return the label for the toggle button.
+   * If there's a selected date, the date is included in the label.
+   */
+  private getToggleButtonLabel(day: DayModel): string {
+    if (day) {
+      const formattedDate = this.dateIOService.toLocaleDisplayFormatString(day.toDate());
+
+      return (
+        this.commonStrings.parse(this.commonStrings.keys.datepickerToggleChangeDateLabel, {
+          SELECTED_DATE: formattedDate,
+        }) || this.commonStrings.keys.datepickerToggle
+      );
+    }
+    return this.commonStrings.keys.datepickerToggleChooseDateLabel || this.commonStrings.keys.datepickerToggle;
+  }
+
+  private listenForDateChanges() {
```

Comment:
[This line](https://github.com/vmware/clarity/blob/angular/projects/angular/src/forms/datepicker/date-input.ts#L119) is the reason why we get the `ExpressionChangedAfterItHasBeenCheckedError` error.